src/NuGet.Clients/NuGet.SolutionRestoreManager/RestoreOperationLogger.cs

```diff
@@ -25,7 +25,7 @@ namespace NuGet.SolutionRestoreManager
     /// Aggregates logging and UI services consumed by the <see cref="SolutionRestoreJob"/>.
     /// </summary>
     [Export]
-    [PartCreationPolicy(CreationPolicy.NonShared)]
+    [PartCreationPolicy(CreationPolicy.Shared)]
```

Comment:
Given this is a disposable object and it was "nonshared" before is there any concern how it was used earlier? If components that imported it, disposed their own instance with the belief that it is not shared they would not dispose the singleton instance.

The object also doesn't seem to support parallel use (ie can only handle one operation at a time due to its start/stop design)

Reply:
> Upon further look, the internal usage would actually dispose the logger here:
> 
> https://github.com/NuGet/NuGet.Client/blob/02e2226188e4e08ca872aa801c6f7e09baed8c72/src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreWorker.cs#L607

@BertanAygun that dispose was not really working. It looks like with MEF things doesn't dispose same way as other items, similar things exceptions also exist with HttpClient(HttpClient instances doesn't simply dispose with dispose).

Reply:
Dispose should be called on the logger object (GetService returns the actual instance) but as you said in this case it doesn't end up doing anything since the code in logger disposes a cancellation token source that is set again next time StartAsync is called but I still think it is false to rely on that assumption. 

Given the object is a singleton now, it would be better to not use it in a using statement anymore to avoid some future issues where dispose is changed to do more.

Reply:
If this is not going to be shared, then we should not request it from the component model every time. 

The SolutionRestoreWorker should own it. 

Maybe that resolves the symptoms that @zivkan was seeing.

Reply:
Also @erdembayar make sure you look at https://github.com/microsoft/vs-mef/issues/135

Reply:
@zivkan @nkolev92 please check now. I'm not touching NonShared part this time and that HashSet keep holding reference is gone. 
Thank you.